pkg/client/rootd/dns/server.go

```diff
@@ -249,6 +260,24 @@ func (s *Server) resolveInCluster(c context.Context, q *dns.Question) (result dn
 	return result, rCode, nil
 }
 
+func (s *Server) ResolveAlias(query string) *string {
+	for i := range s.config.Mappings {
+		mappingName := s.config.Mappings[i].Name + "."
+		if strings.Count(mappingName, ".") == 1 {
+			mappingName += s.clusterDomain
+		}
+
+		if mappingName == query {
+			dotName := s.config.Mappings[i].AliasFor + "."
+			if strings.Count(dotName, ".") == 1 {
+				dotName += s.clusterDomain
+			}
```

Comment:
Same question here.

Reply:
Related to thing before, but that's probably not necessary.